ge/org.osate.ge/src/org/osate/ge/internal/businessObjectHandlers/FlowSpecificationInstanceHandler.java

```diff
@@ -0,0 +1,41 @@
+package org.osate.ge.internal.businessObjectHandlers;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+import javax.inject.Named;
+
+import org.osate.aadl2.Feature;
+import org.osate.aadl2.FlowEnd;
+import org.osate.aadl2.instance.FeatureInstance;
+import org.osate.aadl2.instance.FlowSpecificationInstance;
+import org.osate.ge.di.GetName;
+import org.osate.ge.di.Names;
+
+class FlowSpecificationInstanceHandler {
+	@GetName
+	public String getName(final @Named(Names.BUSINESS_OBJECT) FlowSpecificationInstance fsi) {
+		return fsi.getName();
+	}
+
+	/**
+	 * Gets an array of business objects which describes the logical diagram element path to the flow end.
+	 * @param fsi
+	 * @param flowEnd
+	 * @param addFeatureInstance - get the source or destination FeatureInstance of the FlowSpecificationInstance
+	 * @return
+	 */
+	protected static Object[] getBusinessObjectsPathToFlowEnd(final FlowSpecificationInstance fsi,
```

Comment:
Rework this method. Adjust ConnectionReferenceEnd.getBusinessObjectsPathToConnectionInstanceEnd to be more generic so it can be used and moved to a Util class. Let's discuss if there is a reason for going to declarative model to handle feature groups but that shouldn't be necessary.